lib/trino-aws/src/main/java/io/trino/aws/AwsCredentialsProviderConfig.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.aws;
+
+import java.util.Optional;
+
+public interface AwsCredentialsProviderConfig
```

Comment:
The way it is now is to require the actual configuration class to implement this interface. This way the actual property names are unchanged, but we could probably do an additional unification of those property names (with different prefixes) so that they are consistent between uses.

Reply:
I would prefer to define config here and reuse `io.airlift.configuration.ConfigBinder#bindConfig(java.lang.Class<T>, java.lang.Class<? extends java.lang.annotation.Annotation>, java.lang.String)`

WDYT?

Reply:
I pushed a version implementing this, without the S3 code changes.